packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -334,6 +372,48 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
     this.subscribeDeviceManagerEvents();
 
     this.callClient.onStateChange(onStateChange);
+    /* @conditional-compile-remove(call-transfer) */
+    if (this.callAgent.kind === 'CallAgent') {
+      const onCallsUpdated = (args: { added: Call[]; removed: Call[] }): void => {
+        if (this.call?.id) {
+          const removedCall = args.removed.find((call) => call.id === this.call?.id);
+          if (removedCall) {
+            const removedCallState = this.callClient.getState().callsEnded[removedCall.id];
+            const latestAcceptedTransfer = findLatestAcceptedTransfer(
+              removedCallState.transferFeature.acceptedTransfers
+            );
+            const _callAgent = callAgent as CallAgent;
+            const transferCall = _callAgent.calls.find(
+              (call: Call) => call.state === 'Connected' && call.id === latestAcceptedTransfer?.callId
```

Comment:
I would also wonder if there is a chance that when this fires that it might find the original call if it goes too soon?

Reply:
Good point. I will follow up with calling if this is possible. But I removed the condition for the transfer call to be Connected because it may still be in Connecting which we still want to switch to when the old call is removed.

Reply:
@dmceachernmsft if I understand your question correctly it should not find original call because I am looking for the latestAcceptedTransfer call id which should be different